x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/data_tier_allocation_field/components/no_tiers_available_notice.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { FunctionComponent } from 'react';
+import { EuiCallOut } from '@elastic/eui';
+
+import { PhaseWithAllocation } from '../../../../../../../../../common/types';
+
+const i18nTexts = {
+  warm: {
+    title: i18n.translate('xpack.indexLifecycleMgmt.warmPhase.dataTier.noTiersAvailableTitle', {
+      defaultMessage: 'No nodes assigned to the warm tier',
```

Comment:
> it's really "No nodes assigned to any tier"

I think my PR description was a bit misleading so I've tried to clarify. The condition is a little more nuanced. Here's an example to clarify.

Let's say the user has nodes assigned to the cold tier but not the warm tier. If the user configures the warm phase to allocate the corresponding tier, it will look for the presence of warm nodes first and then hot nodes as a fallback. In the absence of those nodes, we'll show this message. In this case, saying "No nodes assigned to any tier" would be inaccurate because nodes are assigned to the cold tier, though they are ineligible for allocation of data in the warm phase.